hydra/fileformat.go

```diff
@@ -62,6 +63,10 @@ var (
 	convertFloat FloatConverter
 )
 
+var (
+	hostname, hostnameErr = os.Hostname()
```

Comment:
User will not notice an error `os.Hostname()` returned. 
I think this code should be placed in `NewInTail()`.
If `config.HostnameKey` is defined but `os.Hostname()` returns an error, `NewInTail()` should return an error.